pkg/reloader/reloader.go

```diff
@@ -0,0 +1,212 @@
+// Package reloader contains helpers to trigger reloads of Prometheus instances
+// on configuration changes and to substitude environment variables in config files.
+package reloader
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"hash"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/a8m/envsubst"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	fsnotify "gopkg.in/fsnotify.v1"
+)
+
+// Reloader can watch config files and trigger reloads of a Prometheus server.
+// It optionally substitutes environment variables in the configuration.
```

Comment:
From this point I have no idea how example "substitute config" can look like and what is for.

What file I should use to configure Prom? `cfgSubstFilename` or `cfgSubstFilename`? Also, maybe let's point to the `https://github.com/a8m/envsubst` lib we use for docs, or maybe just shortly add mention how to use it.